rotkehlchen/api/rest.py

```diff
@@ -5431,3 +5431,35 @@ def get_historical_netvalue(
             result['last_event_identifier'] = last_event_id
 
         return api_response(_wrap_in_ok_result(result=result))
+
+    @async_api_call()
+    def get_historical_prices_per_asset(
+            self,
+            asset: Asset,
+            interval: int,
+            to_timestamp: Timestamp,
+            from_timestamp: Timestamp,
+    ) -> dict[str, Any]:
+        prices, no_prices_ts = {}, []
+        with (db := self.rotkehlchen.data.db).conn.read_ctx() as cursor:
+            main_currency = db.get_setting(cursor=cursor, name='main_currency')
+
+        current_ts = from_timestamp
+        while current_ts <= to_timestamp:
+            try:
+                price = PriceHistorian.query_historical_price(
+                    from_asset=asset,
+                    to_asset=main_currency,
+                    timestamp=current_ts,
+                )
+            except NoPriceForGivenTimestamp:
+                no_prices_ts.append(current_ts)
+            else:
+                prices[current_ts] = str(price)
+
+            current_ts += interval  # type: ignore[assignment]  # they're both integers
+
+        return _wrap_in_ok_result(result={
+            'prices': prices,
+            'no_prices_timestamps': no_prices_ts,
```

Comment:
I don't say it's wrong but why choose this approach? Did you talk with the frontend? They prefer this to a single dict with `None` values if missing?